### PR TITLE
Add `observer` to peer info

### PR DESCRIPTION
**Pull request by neilalexander**
This is to support nats-io/nats-server#4582.

Signed-off-by: Neil Twigg <neil@nats.io>

### CODE REVIEW

**Review by ripienaar — APPROVED** [review]
LGTM

File: api/cluster.go [hunk 1]
```diff
@@ -66,9 +66,10 @@ type ClusterInfo struct {
...
 // are supporting the stream or consumer.
 type PeerInfo struct {
-	Name    string        `json:"name" yaml:"name"`
-	Current bool          `json:"current" yaml:"current"`
-	Offline bool          `json:"offline,omitempty" yaml:"offline"`
-	Active  time.Duration `json:"active" yaml:"active"`
-	Lag     uint64        `json:"lag,omitempty" yaml:"lag"`
+	Name     string        `json:"name" yaml:"name"`
+	Current  bool          `json:"current" yaml:"current"`
+	Observer bool          `json:"observer" yaml:"observer"`
```
Comment (ripienaar) [inline api/cluster.go:71]: I noticed you did omitempty in json on the nats.go so we should probably do the same here? Assuming thats whats in the server.  But not on yaml.

**Review by ripienaar — CHANGES_REQUESTED** [review]
Updating review after json omitempty comment

File: api/cluster.go (hunk 1 above)
Comment (neilalexander) [reply api/cluster.go]: Yes good catch, fixed.

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**